Wrong error location when casting pointer to slice (stage2)

### Zig Version

0.10.0-dev.3900+ab4b26d8a

### Steps to Reproduce

Compile

```zig
const AnObject = struct {
    f1: i32 = 0,
    f3: i32 = 0,
};

pub fn main() void {
    var o = AnObject{};
    var ptr = @ptrCast([*]u8, @alignCast(@alignOf([*]u8), &o));
    var slice = ptr[0..];
    _ = slice;
}
```


### Expected Behavior

Error message should point to ` var slice = ptr[0..];`

```
$ /opt/zig/zig build-exe a.zig -fstage1
./a.zig:10:20: error: slice of pointer must include end value
    var slice = ptr[0..];
```


### Actual Behavior

```
$ /opt/zig/zig build-exe a.zig 
a.zig:2:1: error: slice of pointer must include end value
const AnObject = struct {
^~~~~
```

In multi-file project error location is even more confusing:

```
$ /opt/zig/zig build-exe server.zig 
server-sm/worker.zig:2:1: error: slice of pointer must include end value
const std = @import("std");
^~~~~
server-sm/worker.zig:2:1: error: slice of pointer must include end value
const std = @import("std");
^~~~~
```

--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -27202,7 +27202,7 @@ fn analyzeSlice(
         if (!end_is_len) {
             break :e try sema.coerce(block, Type.usize, uncasted_end_opt, end_src);
         }
-        return sema.fail(block, end_src, "slice of pointer must include end value", .{});
+        return sema.fail(block, src, "slice of pointer must include end value", .{});
     };
 
     const sentinel = s: {

--- a/test/cases/compile_errors/slice_of_pointer_must_include_end_value.zig
+++ b/test/cases/compile_errors/slice_of_pointer_must_include_end_value.zig
@@ -1,0 +1,10 @@
+comptime {
+    var ptr: [*]u8 = undefined;
+    _ = ptr[0..];
+}
+
+// error
+// backend=stage2
+// target=native
+//
+// :3:12: error: slice of pointer must include end value

